runtime/doc/pi_health.txt

```diff
@@ -23,60 +23,135 @@ Commands				*health-commands*
+Lua Functions				*health-functions-lua* *health-lua*
+
+health.vim Lua functions are for creating new healthchecks. They mostly just
+do some layout and formatting, to give users a consistent presentation. See
+also |health-functions-vim|
+
+The Lua health module provides the functions:  >
+	local health = require('health')
+<
+health.report_start({name})				*health.report_start()*
+	Starts a new report. Most plugins should call this only once, but if
+	you want different sections to appear in your report, call this once
+	per section.
+
+health.report_info({msg})				*health.report_info()*
+	Reports an informational message.
+
+health.report_ok({msg})					*health.report_ok()*
+	Reports a "success" message.
+
+health.report_warn({msg} [, {advice}])			*health.report_warn()*
+	Reports a warning. {advice} is an optional List of suggestions.
+
+health.report_error({msg} [, {advice}])			*health.report_error()*
+	Reports an error. {advice} is an optional List of suggestions.
+
 ==============================================================================
-Functions				*health-functions*
+Create a Lua healthcheck		*health-dev-lua*
+
+Healthchecks are functions that check the user environment, configuration,
+etc. Nvim has built-in healthchecks in $VIMRUNTIME/autoload/health/.
+
+To add a new healthcheck for your own plugin, simply define a Lua module in
+your plugin that returns a table with a function check(), |:checkhealth|
+automatically finds and invokes such modules function.
+
+If your plugin is named "foo", then its healthcheck module should be a file in
+one of this locations on the 'runtimepath':
+	- `lua/foo/health/init.lua`
+	- `lua/foo/health.lua`
+
+If your plugin provides a submodule named "bar" and you want to have a separate
+checkhealth for it you can define the healthcheck at one of this locations also
+on the 'runtimepath':
+	- `lua/foo/bar/health/init.lua`
+	- `lua/foo/bar/health.lua`
+
+All submodules should return a Lua table containing the method `check()`
+
+Copy this sample code into `lua/foo/health/init.lua` or `lua/foo/health.lua`
+replacing "foo" in the path with your plugin name: >
+
+	local M = {}
+	local health = require("health")
+
+	M.check = function()
+	  health.report_start("my_plugin report")
+	  -- make sure setup function parameters are ok
+	  if check_setup() then
+	    health.report_ok("Setup function is correct")
+	  else
+	    health.report_error("Setup function is incorrect")
+	  end
+	  -- do some more checking
+	  -- ...
+	end
+
+	return M
+
+==============================================================================
+Vimscript Functions		*health-functions-vimscript* *health-vimscript*
 
 health.vim functions are for creating new healthchecks. They mostly just do
-some layout and formatting, to give users a consistent presentation.
+some layout and formatting, to give users a consistent presentation. See also
+|health-functions-lua|
```

Comment:
Same comment as above, I don't think we need to mention that these functions just do layout and formatting. It's not relevant to the user.